### PR TITLE
Auto-close dm / gm channels

### PR DESCRIPTION
**Pull request by ccbrown**
#### Summary
Auto-closes DM/GM channels after 7 days with no new messages.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-7984

#### Checklist
- [x] Added or updated unit tests (required for all new features)

#### Test Information
This PR was tested on: Chrome, MacOS

### CODE REVIEW

File: src/selectors/entities/channels.js [hunk 1]
```diff
@@ -222,35 +222,37 @@ export const getOtherChannels = createSelector(
...
-    (currentChannelId, channels, myPreferences, teammateNameDisplay, usersState) => {
+    (currentChannelId, channels, config, myPreferences, teammateNameDisplay, usersState) => {
         const allChannels = channels.map((c) => {
             const channel = {...c};
             channel.isCurrent = c.id === currentChannelId;
             return channel;
         });
 
-        return buildDisplayableChannelList(usersState, allChannels, myPreferences, teammateNameDisplay);
+        return buildDisplayableChannelList(usersState, allChannels, config, myPreferences, teammateNameDisplay);
```
Comment (enahum) [inline src/selectors/entities/channels.js]: I think this is ok for now but actually we are trying to move away from this functions as they are really poor on performance

File: src/selectors/entities/channels.js [hunk 2]
```diff
@@ -222,35 +222,37 @@ export const getOtherChannels = createSelector(
 export const getChannelsByCategory = createSelector(
     getCurrentChannelId,
     getMyChannels,
+    getConfig,
```
Comment (enahum) [inline src/selectors/entities/channels.js]: perhaps think of passing just the property from the config that you need, that way if something else in the config changes the selector might not need to re-calculate

**Comment by enahum** [comment]
One more thing, what happens with the autoclose preference when you manually close a DM?

**Comment by ccbrown** [comment]
Will update `getSortedDirectChannelIds`.

When you manually close the DM, the "CATEGORY_DIRECT_CHANNEL_SHOW" preference is set to false, forcing it to hide until you open it again or you get another message on it. Should work as expected.

**Comment by ccbrown** [comment]
@enahum Take another look? Updated `getSortedDirectChannelIds`. Unfortunately, I think re-calculations are gonna have to be done pretty regularly as the the displayed channels can change on new posts.

**Pull request merged**